client/app/telecom/pack/xdsl/modem/router/dhcp/pack-xdsl-modem-dhcp.controller.js

```diff
@@ -1,56 +1,107 @@
-angular.module('managerApp').controller('XdslModemDhcpCtrl', function ($stateParams, PackXdslModemDhcpObject, OvhApiXdsl, $q, $translate, TucToast, tucValidator, TucPackXdslModemMediator) {
-  const self = this;
-  this.validator = tucValidator;
-  this.mediator = TucPackXdslModemMediator;
+angular.module('managerApp').controller('XdslModemDhcpCtrl',
+  function ($q, $stateParams, $translate, $uibModal, OvhApiXdsl, PackXdslModemDhcpObject,
+    TucPackXdslModemMediator, TucToast, tucValidator) {
+    const self = this;
+    this.validator = tucValidator;
+    this.mediator = TucPackXdslModemMediator;
 
-  /**
+    /**
      * submit / unsubmit with keys
      * @param                   {Event} $event AngularJs Event
      * @param {PackXdslModemDhcpObject} dhcp   dhcp to update
      * @param                 {Boolean} valid  Form valid ?
      */
-  this.watchKey = function ($event, dhcp, valid) {
-    if ($event.keyCode === 13 && valid) {
-      self.submit(dhcp);
-    }
-    if ($event.keyCode === 27) {
-      dhcp.cancel();
-    }
-  };
+    this.watchKey = function ($event, dhcp, valid) {
+      if ($event.keyCode === 13 && valid) {
+        self.submit(dhcp);
+      }
+      if ($event.keyCode === 27) {
+        dhcp.cancel();
+      }
+    };
 
-  /**
+    /**
      * Submit a DHCP
      * @param {PackXdslModemDhcpObject} lan LAN to update
      * @return {Promise}
      */
-  this.submit = function (dhcp) {
-    return dhcp.save($stateParams.serviceName);
-  };
-
-  this.isIpInOrder = function (ip1Param, ip2Param) {
-    let ip1 = ip1Param;
-    let ip2 = ip2Param;
-    ip1 = _.map(ip1.split(/\./), elt => parseInt(elt, 10));
-    ip2 = _.map(ip2.split(/\./), elt => parseInt(elt, 10));
-    const comp = _.reduce(ip1, (result, val, index) => result && (val <= ip2[index]));
-
-    return comp && _.last(ip1) < _.last(ip2);
-  };
-
-  function init() {
-    self.loader = true;
-    return OvhApiXdsl.Modem().Lan().Dhcp().Aapi()
-      .query({
-        xdslId: $stateParams.serviceName,
-      }).$promise.then((data) => {
-        self.dhcps = _.map(data, elt => new PackXdslModemDhcpObject(elt));
-      }).catch((err) => {
-        TucToast.error($translate.instant('xdsl_modem_dhcp_error'));
-        return $q.reject(err);
-      }).finally(() => {
-        self.loader = false;
-      });
-  }
-
-  init();
-});
+    this.submit = function (dhcp) {
+      return dhcp.save($stateParams.serviceName);
+    };
+
+    this.isIpInOrder = function (ip1Param, ip2Param) {
+      let ip1 = ip1Param;
+      let ip2 = ip2Param;
+      ip1 = _.map(ip1.split(/\./), elt => parseInt(elt, 10));
+      ip2 = _.map(ip2.split(/\./), elt => parseInt(elt, 10));
+      const comp = _.reduce(ip1, (result, val, index) => result && (val <= ip2[index]));
+
+      return comp && _.last(ip1) < _.last(ip2);
+    };
+
+    this.checkIP = function (ipParam) {
+      let ipToUse = null;
+      switch (ipParam) {
+        case '91.121.161.184':
```

Comment:
Shouldn't be handled directly on the front side.

Is there any issue to provide these values through an API?

Reply:
Theses kind of data can't be managed in frontend, all more in a controller. (don't forget constant :) )

For example, we shouln't have to redeploy or rebuild our app if this data is updated. So, It should be managed on API side.

We can't approve this pull request as long as the data will be inside.

Feel free to contact me if needed @Steffy29

Reply:
Little forgetfulness, from a frontend point of view:

checkIP method should be managed in a service, not in a controller.

For example, if you have 2 controllers using a "checkIp", you don't want to develop this function two times.

So, if you have to manage this kind of case in the frontend (not here, should be managed in API), keep in mind this pattern:

controller A > Service method > constant 
controller B > Service method > constant

With this example: 

controller DHCP > Service IP (checkIp method) > constant (list of ips)

See you :)